inference-engine/src/transformations/src/transformations/op_conversions/convert_convertlike.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/op_conversions/convert_convertlike.hpp"
+
+#include <memory>
+#include <ngraph/opsets/opset8.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+#include <ngraph/rt_info.hpp>
+#include <vector>
+
+#include "itt.hpp"
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::ConvertConvertLike, "ConvertConvertLike", 0);
+
+using namespace ngraph;
+
+ngraph::pass::ConvertConvertLike::ConvertConvertLike() {
+    MATCHER_SCOPE(ConvertConvertLike);
+
+    auto convertlike = pattern::wrap_type<opset8::ConvertLike>();
+
+    matcher_pass_callback callback = [](pattern::Matcher& m) {
+        auto cvtlike = std::dynamic_pointer_cast<opset8::ConvertLike>(m.get_match_root());
+        if (!cvtlike) {
+            return false;
+        }
+
+        auto like = cvtlike->input_value(1);
+        const element::Type& dest_type = like.get_element_type();
+        if (dest_type == element::dynamic || dest_type == element::undefined)
+            return false;
```

Comment:
I don't think we need this check. If plugin receives this undefined or dynamic type even with ConvertLike it will throw an error.

Reply:
Moreover Convert can propagate any of this element types.